pytest_fixtures/broker.py

```diff
@@ -43,6 +43,15 @@ def factory(retry_limit=3, delay=300, **broker_args):
     return factory
 
 
+@pytest.fixture
+def param_host(request):
+    """A function-level fixture that provides a host object based broker args"""
+    if isinstance(request.param, dict):
+        nick = request.param.pop('nick')
```

Comment:
I don't think there's any need to separate nick from the rest of the keys in the dictionary, and just pass `**request.param` to VMBroker.

This is a good start to a more abstract broker fixture, but puts the requirement on the test to parametrize everything without providing any defaults.

`param_host` doesn't give much context within the scope of the tests that are using it, and should provide some context to differentiate the types of content hosts that we'll be providing through broker.  I'd suggest something like `rhel_vm`, which at least shows that its a rhel virtual machine. We'll be parametrizing attributes of that rhel vm, like its version, compute resources, boot config (FIPS), and count. We'll also be building `rhel_containers`, and who knows what other OS or host types.

The options and exact parameter values for these parametrization matrixes for these fixtures should be decoupled from the test framework through the settings, so that they're not hardcoded in any way. Test functions, classes, and modules need to be able to limit some of these parameters - but only to what's within scope of the available configuration.